### PR TITLE
Make more python functions private

### PR DESCRIPTION
**Pull request by SouthEndMusic**
Fixes https://github.com/Deltares/Ribasim/issues/1651

### CODE REVIEW

**Review by visr — APPROVED** [review]
Looks good to me, once these are resolved:

```
ribasim_qgis/core/nodes.py:108: error: "QgsVectorLayer" has no attribute "_fields"; maybe "fields"?  [attr-defined]
ribasim_qgis/core/nodes.py:11[6](https://github.com/Deltares/Ribasim/actions/runs/10300476122/job/28509943889?pr=1702#step:7:7): error: "QgsVectorLayer" has no attribute "_fields"; maybe "fields"?  [attr-defined]
ribasim_qgis/core/nodes.py:188: error: "QgsVectorLayer" has no attribute "_fields"; maybe "fields"?  [attr-defined]
ribasim_qgis/core/topology.py:52: error: "QgsVectorLayer" has no attribute "_fields"; maybe "fields"?  [attr-defined]
ribasim_qgis/core/topology.py:116: error: "QgsVectorLayer" has no attribute "_fields"; maybe "fields"?  [attr-defined]
```

**Comment by visr** [comment]
Does this fully fix https://github.com/Deltares/Ribasim/issues/1651 though? I think our public API can be smaller still.

File: ribasim_qgis/core/topology.py [hunk 1]
```diff
@@ -49,7 +49,7 @@ def collect_node_properties(
     node: QgsVectorLayer,
 ) -> tuple[NDArray[np.float64], NDArray[np.int_], dict[str, tuple[str, int]]]:
     n_node = node.featureCount()
-    node_fields = node.fields()
+    node_fields = node._fields()
```
Comment (evetion) [inline ribasim_qgis/core/topology.py:52]: QGIS =/= Ribasim, so these should not be renamed (this is a different function altogether)

File: ribasim_qgis/core/topology.py [hunk 2]
```diff
@@ -113,7 +113,7 @@ def set_edge_properties(node: QgsVectorLayer, edge: QgsVectorLayer) -> None:
     edge_xy = collect_edge_coordinates(edge)
     from_fid, to_fid = derive_connectivity(node_index, node_xy, edge_xy)
 
-    edge_fields = edge.fields()
+    edge_fields = edge._fields()
```
Comment (evetion) [inline ribasim_qgis/core/topology.py:116]: Same here

**Review by evetion — CHANGES_REQUESTED** [review]

**Review by evetion — APPROVED** [review]

**Pull request merged**